src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1877,6 +1877,10 @@ public void EnsureBGCRevisitInfoAlloc()
         /// </summary>
         public int HeapCount { get; private set; } = -1;
         /// <summary>
+        /// Number of Generations (0,1,2,LOH,[POH]), which is either 4 or 5 depending on version
+        /// </summary>
+        public int GenerationCount { get { return this.PerHeapHistories[0].GenData.Length; } }
```

Comment:
From a review of the existing code in this file, other accesses of `PerHeapHistories` is guarded with a `null` check and validation that the length of the array is non-zero.  Can you add this here as well please?

Reply:
```suggestion
        public int GenerationCount { get { return this.PerHeapHistories?.FirstOrDefault()?.GenData.Length ?? 0; } }
```